mne_bids/dataframe.py

```diff
@@ -0,0 +1,134 @@
+import numpy as np
+from collections import OrderedDict as odict
+
+
+class DataFrame():
+    def __init__(self, data):
+        """ A stripped down clone of a Pandas DataFrame object
+
+        Parameters
+        -----------
+        data : ordered dictionary
+            An ordered dictionary containing the column names (keys) and
+            associated column values (values).
+
+        Note: All data is cast as a string to ensure uniformity and to avoid
+        issues when comparing data. Because of this, operations on the data
+        in this format should be avoided and any processing done before
+        assignment.
```

Comment:
can you maybe make the data read only? Example using this: https://docs.scipy.org/doc/numpy-1.15.1/reference/generated/numpy.ndarray.flags.html

Reply:
Ok, I made the data attribute a property with no write method which is the best you can do in python